lib/matplotlib/patches.py

```diff
@@ -1462,6 +1462,93 @@ def get_patch_transform(self):
         return self._patch_transform
 
 
+class Annulus(Patch):
+    """
+    An elliptical annulus.
+    """
+
+    def __str__(self):
+        if self.a == self.b:
+            r = str(self.a)
+        else:
+            r = '(%s, %s)' % (self.a, self.b)
+
+        return "Annulus(xy=(%s, %s), r=%s, width=%s, angle=%s)" % \
+                (self.center[0], self.center[1], r, self.width, self.angle)
+
+    @docstring.dedent_interpd
+    def __init__(self, xy, r, width, angle=0.0, **kwargs):
+        """
+        xy : array_like
+            center of annulus
+
+        r : scalar or 1D array_like
+            The radius, or semi-major axes
+                - If float: radius of the outer circle
+                - If array_like of size 2: semi-major and -minor axes of outer ellipse
+
+        width : float
+          width of the annulus
+
+        angle: float, optional
+          rotation in degrees (anti-clockwise). Ignored for circular annuli (`r` is a scalar)
+
+
+        Valid kwargs are:
+        %(Patch)s
+        """
+        Patch.__init__(self, **kwargs)
+
+        if np.size(r) == 2:
+            self.a, self.b = r
+        elif np.size(r) == 1:
+            self.a = self.b = float(r)
+        else:
+            raise ValueError(
+                'r parameter should be either float, or array_like of size 2')
```

Comment:
This message is misleading, as it accepts `r=[[[1]]]` just fine. Perhaps you meant `np.shape(r) == ()` and `np.shape(r) == (2,)` as the conditions?